go/indexer/services/l1/bridge/filter.go

```diff
@@ -20,13 +20,12 @@ func FilterStateBatchAppendedWithRetry(filterer *scc.StateCommitmentChainFiltere
 		ctxt, cancel := context.WithTimeout(opts.Context, DefaultConnectionTimeout)
 		opts.Context = ctxt
 		res, err := filterer.FilterStateBatchAppended(opts, nil)
-		switch err {
-		case nil:
+		if err == nil {
 			cancel()
-			return res, err
-		default:
-			logger.Error("Error fetching filter", "err", err)
+			return res, nil
 		}
+		cancel()
```

Comment:
we're canceling on both code paths. It would be simpler to cancel right after L22 and remove the `cancel()` call here and on L24.